File: common/internal_model/src/main/scala/weco/catalogue/internal_model/locations/AccessCondition.scala
package weco.catalogue.internal_model.locations

case class AccessCondition(
  method: AccessMethod,
  status: Option[AccessStatus] = None,
  note: Option[String] = None,
  terms: Option[String] = None
) {
  def isEmpty: Boolean =
    this == AccessCondition(method = AccessMethod.NotRequestable)

  def isAvailable: Boolean = status.exists(_.isAvailable)

  def hasRestrictions: Boolean = status.exists(_.hasRestrictions)
}

case object AccessCondition {
  def apply(method: AccessMethod, status: AccessStatus): AccessCondition =
    AccessCondition(method = method, status = Some(status))
}

File: catalogue_graph/src/models/pipeline/access_condition.py
from models.pipeline.access_method import AccessMethod
from models.pipeline.access_status import AccessStatus
from models.pipeline.serialisable import SerialisableModel


class AccessCondition(SerialisableModel):
    method: AccessMethod
    status: AccessStatus | None = None
    terms: str | None = None
    note: str | None = None

    @property
    def is_available(self) -> bool:
        return self.status is not None and self.status.is_available

    @property
    def has_restrictions(self) -> bool:
        return self.status is not None and self.status.has_restrictions
